Idle time, connection TTL and other features

droscy opened the pull request:
Hello, I started receiving invalid data from inverter after a firmware upgrade and so I added to modbus-proxy some features in order to diagnose and bypass inconsistent reading.

These are the new parameters I've added:
- `idle_time`: how many seconds to wait before closing an idle connection to the modbus device
- `connection_ttl`: max duration of a connection, in seconds, to the modbus device
- `reconnect_delay`: how many seconds to wait before establishing a new connection to the modbus device after an error
- `request_delay`: how many seconds to wait between two requests

droscy committed:
Add an idle-tracker to close idle modbus connection

droscy committed:
Review connection close in idle-tracker

droscy committed:
Reformat code

droscy committed:
Add parameter reconnect_delay

droscy committed:
Add parameter request_delay

droscy committed:
Add ttl-monitor to close modbus connection after a chosen time

droscy committed:
Refactor after flake8 checks

droscy committed:
Update README with new parameters

tiagocoutinho commented on README.md:
```diff
@@ -43,6 +43,10 @@ devices:
+    connection_ttl: 10         # max duration of the connection (m) (optional, default: 0, disabled)
+    idle_time: 5               # max idle time of the connection (s) (optional, default: 0, disabled)
+    reconnect_delay: 1         # delay reconnection after error (s) (optional, default: 0)
+    request_delay: 100         # delay each request (ms) (optional, default: 0)
```
I would prefer to keep all time units in seconds for coherence

droscy replied:
Yes, it is better. Changes committed.

tiagocoutinho commented on src/modbus_proxy.py:
```diff
@@ -25,7 +26,7 @@
         "console": {"class": "logging.StreamHandler", "formatter": "standard"}
     },
-    "root": {"handlers": ["console"], "level": "INFO"},
+    "root": {"handlers": ["console"], "level": "DEBUG"},
```
Let's keep the default level "INFO"

droscy replied:
Sure, I forgot it.

droscy committed:
Rollback default logging level

droscy committed:
Change all time units in seconds

tiagocoutinho reviewed (changes requested):
Looks good.
Sorry for the late feedback.
I would feel more comfortable if the PR included tests. Would you mind adding a few?

droscy commented:
> I would feel more comfortable if the PR included tests. Would you mind adding a few?

Sure, as soon as I can study your test framework I'll add some test cases.

droscy committed:
Add Eclipse settings to gitignore

droscy committed:
Add tests for new parameters

droscy committed:
Change contributors

droscy commented:
I've added some tests for the new parameters. All these tests are based on the `asyncio.sleep()` function to emulate the passing of time, so there is a remote chance of failure because the `sleep()` is not 100% time-accurate.

I've also added myself among contributors; other contributors are present in git history but I left the decision to you to add them to the readme file.
